tests/musig_tests.rs

```diff
@@ -0,0 +1,197 @@
+#[cfg(test)]
+mod musig_tests {
+    use bellman::pairing::bn256::Bn256;
+    use franklin_crypto::alt_babyjubjub::{AltJubjubBn256, FixedGenerators};
+    use franklin_crypto::eddsa::{PrivateKey, PublicKey, Seed};
+    use musig::musig::{MusigSession, MusigVerifier};
+    use musig::musig_hash::Sha256HStar;
+    use rand::{thread_rng, Rng};
+
+    struct SplitIterator<'a, T> {
+        index: usize,
+        left: &'a mut [T],
+        right: &'a mut [T],
+    }
+
+    impl<'a, T> SplitIterator<'a, T> {
+        fn new(left: &'a mut [T], right: &'a mut [T]) -> SplitIterator<'a, T> {
+            SplitIterator {
+                index: 0,
+                left,
+                right,
+            }
+        }
+
+        fn next(&mut self) -> Option<&mut T> {
+            let res;
+
+            if self.index < self.left.len() {
```

Comment:
You can assign a result of `if` statements in Rust:

```rust
let value = if true {
    42
} else {
    24
};
```

Reply:
Fixed